test/e2e/authorizer/authorizer_test.go

```diff
@@ -0,0 +1,417 @@
+	}
+
+	var testCases = []struct {
+		name string
+		work func(ctx context.Context, t framework.TestingTInterface, server testSetup)
+	}{
+		{
+			name: "create namespace and configmap as user-1",
+			work: func(ctx context.Context, t framework.TestingTInterface, setup testSetup) {
+				createResources(ctx, "resources.yaml", setup.clients["admin"].KubeClient, setup.clients["admin"].KcpClient)
+
+				_, err = setup.clients["user-1"].KubeClient.CoreV1().Namespaces().Create(
+					ctx,
+					&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}},
+					metav1.CreateOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to create namespace: %v", err)
+					return
+				}
+
+				_, err = setup.clients["user-1"].KubeClient.CoreV1().ConfigMaps("default").Create(
+					ctx,
+					&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+					metav1.CreateOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to create configmap: %v", err)
+					return
+				}
+			},
+		},
+		{
+			name: "allow view deny creation of resources for user-2",
+			work: func(ctx context.Context, t framework.TestingTInterface, server testSetup) {
+				createResources(ctx, "resources.yaml", server.clients["admin"].KubeClient, server.clients["admin"].KcpClient)
+
+				_, err = server.clients["user-1"].KubeClient.CoreV1().Namespaces().Create(
+					ctx,
+					&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "default"}},
+					metav1.CreateOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to create namespace: %v", err)
+					return
+				}
+
+				_, err = server.clients["user-1"].KubeClient.CoreV1().ConfigMaps("default").Create(
+					ctx,
+					&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+					metav1.CreateOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to create configmap: %v", err)
+					return
+				}
+
+				_, err = server.clients["user-2"].KubeClient.CoreV1().Namespaces().Create(
+					ctx,
+					&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "baz"}},
+					metav1.CreateOptions{},
+				)
+				if err == nil {
+					t.Errorf("expected namespace creation to fail but it didn't")
+					return
+				}
+
+				_, err = server.clients["user-2"].KubeClient.CoreV1().ConfigMaps("default").Create(
+					ctx,
+					&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+					metav1.CreateOptions{},
+				)
+				if err == nil {
+					t.Errorf("expected configmap creation to fail but it didn't")
+					return
+				}
+
+				_, err = server.clients["user-2"].KubeClient.CoreV1().Namespaces().Get(
+					ctx, "default",
+					metav1.GetOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to get namespace: %v", err)
+					return
+				}
+
+				_, err = server.clients["user-2"].KubeClient.CoreV1().ConfigMaps("default").Get(
+					ctx, "foo",
+					metav1.GetOptions{},
+				)
+				if err != nil {
+					t.Errorf("failed to get configmap: %v", err)
+					return
+				}
+			},
+		},
+		{
+			name: "deny anything for user-3",
```

Comment:
Do the deny checks in this test case have to be executed in isolation from the success checks from the previous test case? Maybe add the deny checks to the previous test case?

Reply:
no, let's merge into one test case.